frontend/public/components/alert.tsx

```diff
@@ -0,0 +1,450 @@
+    const alert = _.find(_.get(rule, 'alerts'), {labels});
+    if (alert) {
+      return {alert, name, rule};
+    }
+  }
+  return {alert: null, name, rule: _.find(rules, {name, labels})};
+};
+
+class AlertsDetailsPage_ extends SafetyFirst<AlertsPageProps, any> {
+  props: AlertsPageProps;
+
+  componentDidMount () {
+    loadRules();
+  }
+
+  render () {
+    const {alert, loaded, loadError, name, rule} = this.props;
+
+    const severity = _.get(rule, 'labels.severity');
+    const activeAt = _.get(alert, 'activeAt');
+    const annotations = _.get(alert || rule, 'annotations', {});
+
+    return <React.Fragment>
+      <NavTitle detail={true} kind={alertKind} obj={{data: alert}} title={name} />
+      <StatusBox data={rule} loaded={loaded} loadError={loadError}>
+        <div className="co-m-pane__body">
+          <SectionHeading text="Alert Overview" />
+          <div className="co-m-pane__body-group">
+            <div className="row">
+              <div className="col-sm-6">
+                <dl className="co-m-pane__details">
+                  <dt>Name</dt>
+                  <dd>{name}</dd>
+                  {severity && <React.Fragment>
+                    <dt>Severity</dt>
+                    <dd>{_.startCase(severity)}</dd>
+                  </React.Fragment>}
+                  <dt>State</dt>
+                  <dd>
+                    <State state={alertRuleState(rule)} />
+                    {activeAt && <div className="text-muted monitoring-timestamp">Active since&nbsp;<Timestamp timestamp={activeAt} /></div>}
+                  </dd>
+                  <dt>Alert Rule</dt>
+                  <dd>
+                    <div className="co-resource-link">
+                      <ResourceIcon kind={ruleKind} />
+                      <Link to={detailsURL('AlertRule', name, _.get(rule, 'labels'))} className="co-resource-link__resource-name">{name}</Link>
+                    </div>
+                  </dd>
+                </dl>
+              </div>
+              <div className="col-sm-6">
+                <dl className="co-m-pane__details">
+                  <Annotation title="Description">{annotations.description}</Annotation>
+                  <Annotation title="Summary">{annotations.summary}</Annotation>
+                  <Annotation title="Message">{annotations.message}</Annotation>
+                  {annotations.runbook_url && <React.Fragment>
+                    <dt>Runbook</dt>
+                    <dd><ExternalLink href={annotations.runbook_url} text={annotations.runbook_url} /></dd>
+                  </React.Fragment>}
+                </dl>
+              </div>
+            </div>
+          </div>
+        </div>
+      </StatusBox>
+    </React.Fragment>;
+  }
+}
+export const AlertsDetailsPage = connectPage(connect(alertStateToProps)(AlertsDetailsPage_));
+
+// Convert duration in billionths of a second into a formatted string like "1h 30m"
+const formatDuration = (nanoseconds) => {
```

Comment:
Thanks, I didn't notice that we already had a `formatDuration()` 😄 

The format is slightly different (doesn't include `d` for days and shows zero seconds like `5m 0s` instead of just `5m`), but I think it is OK.